component/stream/stream_cache.go

```diff
@@ -200,3 +210,104 @@ func (c *cache) filePurge(key, value interface{}) {
 	c.blocks.Remove(key)
 	c.evictedBlock = blockKey{}
 }
+
+// Using key construct a file name for persisted block
+func (c *cache) getLocalFilePath(key blockKey) string {
+	return filepath.Join(c.diskPath, key.fileKey+"__"+fmt.Sprintf("%d", key.offset)+"__")
```

Comment:
Will this run into issues if the fileKey is fileName? If we have multiple handles of a file?

Reply:
We will never have multiple handles to the file as we maintain only one block in cache irrespective of number of handles open for that file. Also the stream blocks are in memory while this persistence is done on disk so there is no collision in actual file and persisted block. If your concern is about actual file name also following similar format then that also shall not hit issue as our persisted block will now look something like A__1__.txt__0__ so while reading it back we will again do the same conversion. Also these persisted block files are not visible to user directly through blobfuse only way is to manually list them on the persist directory. Let me know if you have any concern with the naming conventions or if you foresee any issue with this.

Reply:
As Vikas said this should be fine, if the file key is names then regardless of open handles don't have duplicates. If our key is the handle id its still ok since the blocks will have unique id prefix.